### PR TITLE
Build 8.10.4 from source

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xhochy** [comment]
Adopted @isuruf's PR #14 a bit and also got a working `osx-64` build. Sadly CI fails due to out-of-memory issues on prefix replacement of one of the larger files. I'm trying to debug that locally but there it succeeds as I have more RAM. If anyone has an idea how to get CI green with the limited RAM, I'd be happy for that.

**Comment by xhochy** [comment]
The timeout on `osx-64` depends on getting https://github.com/lief-project/LIEF/pull/579 (or a similar approach) merged.

File: recipe/build.sh [hunk 1]
```diff
@@ -48,15 +48,17 @@ pushd source
...
   make HADDOCK_DOCS=NO BUILD_SPHINX_HTML=NO BUILD_SPHINX_PDF=NO -j${CPU_COUNT}
   make HADDOCK_DOCS=NO BUILD_SPHINX_HTML=NO BUILD_SPHINX_PDF=NO install -j${CPU_COUNT}
   # Delete profile-enabled static libraries, other distributions don't seem to ship them either and they are very heavy.
-  find $PREFIX -name '*_p.a' -delete
+  find $PREFIX/lib/ghc-${PKG_VERSION} -name '*_p.a' -delete
+  find $PREFIX/lib/ghc-${PKG_VERSION} -name '*.p_o' -delete
 popd
 
+# Remove ghc from $BUILD_PREFIX to speedup conda-build post-processing
+rm -rf "$BUILD_PREFIX/lib/ghc-${PKG_VERSION}"
```
Comment (mbargull) [inline recipe/build.sh:60]: Is there any reason not to use a custom path for stage0 that does not pollute BUILD_PREFIX?
E.g.
```diff
diff --git a/recipe/build.sh b/recipe/build.sh
index f2e8b53..b619eb9 100644
--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -24,6 +24,8 @@ if [[ "${target_platform}" == linux-* ]]; then
   cp $(which $GCC) $BUILD_PREFIX/bin/$GHC_HOST-gcc
 fi
 
+mkdir stage0
+stage0="$( pwd )/stage0"
 pushd binary
   cp $BUILD_PREFIX/share/gnuconfig/config.* .
   # stage0 compiler: --build=$GHC_BUILD --host=$GHC_BUILD --target=$GHC_BUILD
@@ -36,7 +38,7 @@ pushd binary
     if [[ "${build_platform}" == linux-* ]]; then
       CPP=$BUILD-cpp
     fi
-    LD=$BUILD-ld OBJDUMP=$BUILD-objdump RANLIB=$BUILD-ranlib STRIP=$BUILD-strip ./configure --prefix=$BUILD_PREFIX --with-gmp-includes=$BUILD_PREFIX/include --with-gmp-libraries=$BUILD_PREFIX/lib --build=$GHC_BUILD --host=$GHC_BUILD --target=$GHC_BUILD || (cat config.log; exit 1)
+    LD=$BUILD-ld OBJDUMP=$BUILD-objdump RANLIB=$BUILD-ranlib STRIP=$BUILD-strip ./configure --prefix="${stage0}" --with-gmp-includes=$BUILD_PREFIX/include --with-gmp-libraries=$BUILD_PREFIX/lib --build=$GHC_BUILD --host=$GHC_BUILD --target=$GHC_BUILD || (cat config.log; exit 1)
     make install -j${CPU_COUNT}
   )
 popd
@@ -48,17 +50,17 @@ pushd source
     export CC=$GCC
   fi
   cp $BUILD_PREFIX/share/gnuconfig/config.* .
-  ./configure --prefix=$PREFIX --with-gmp-includes=$PREFIX/include --with-gmp-libraries=$PREFIX/lib --with-ffi-includes=$PREFIX/include --with-ffi-libraries=$PREFIX/lib --target=$GHC_HOST
-  make HADDOCK_DOCS=NO BUILD_SPHINX_HTML=NO BUILD_SPHINX_PDF=NO -j${CPU_COUNT}
-  make HADDOCK_DOCS=NO BUILD_SPHINX_HTML=NO BUILD_SPHINX_PDF=NO install -j${CPU_COUNT}
+  (
+    PATH="${stage0}/bin:${PATH}"
+    ./configure --prefix=$PREFIX --with-gmp-includes=$PREFIX/include --with-gmp-libraries=$PREFIX/lib --with-ffi-includes=$PREFIX/include --with-ffi-libraries=$PREFIX/lib --target=$GHC_HOST
+    make HADDOCK_DOCS=NO BUILD_SPHINX_HTML=NO BUILD_SPHINX_PDF=NO -j${CPU_COUNT}
+    make HADDOCK_DOCS=NO BUILD_SPHINX_HTML=NO BUILD_SPHINX_PDF=NO install -j${CPU_COUNT}
+  )
   # Delete profile-enabled static libraries, other distributions don't seem to ship them either and they are very heavy.
   find $PREFIX/lib/ghc-${PKG_VERSION} -name '*_p.a' -delete
   find $PREFIX/lib/ghc-${PKG_VERSION} -name '*.p_o' -delete
 popd
 
-# Remove ghc from $BUILD_PREFIX to speedup conda-build post-processing
-rm -rf "$BUILD_PREFIX/lib/ghc-${PKG_VERSION}"
-
 #echo "main = putStr \"smalltest\"" > Main.hs
 #ghc -v -O0 -threaded -L$PREFIX/lib -fasm -o smalltest Main.hs
 #./smalltest
```

**Comment by xhochy** [comment]
@conda-forge/ghc This is ready for review. I would handle ppc64le separately as the executable size seems to be more than three times than that on others and thus very hard to fit into CI. 

And I would like to make progress to get an osx-arm64 build with the latest 9.2.1rc 😉

**Comment by xhochy** [comment]
> Thanks a lot for this work. Previously I spent already way too much time on this. For learning, what was the most important that was missing from the previous PRs?

Not sure what that could be, all tiny additions. The only major thing was https://github.com/lief-project/LIEF/pull/579 to get the `osx-*` postprocessing fast enough.

**Pull request merged**